api/src/controllers/team.js

```diff
@@ -313,6 +316,75 @@ async function updateTeam(req, res, next) {
   }
 }
 
+/**
+  * Clear the team Data from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: teamn is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function resetTeamData(req, res, next) {
+  try {
+    const { teamId } = req.params;
+    const { isDeleteFlag } = req.body;
+    await Achievement.deleteMany({ teamId });
+    await Publication.deleteMany({ teamId });
+    if (isDeleteFlag) {
+      await Team.findByIdAndDelete(teamId);
+      return res.status(200).json('Deleted successfully!');
+    } return res.status(200).json('Cleared successfully!');
+  } catch (error) {
+    return next(
+      fillErrorObject(500, 'Error occurred with server', [error.message]),
+    );
+  }
+}
+
+/**
+  * Delete the team from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: team is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function deleteGHPages(req, res, next) {
+  const { ghToken } = req.body;
+  // Call github API to get username
+  const { data } = await axios.get('https://api.github.com/user',
+    {
+      headers: { Authorization: `token ${ghToken}` },
+    });
```

Comment:
You need a try/block here because you don't validate the token before this call.
1. Does the token exist in localStorage? This can be done on the frontend, this way it won't reach this call if there is no token.
2. If the token has expired/is not valid, this will fail, hence you need a try/catch here.

Reply:
Also because we're seeing this block of logic being duplicated, might be good to pull this up into the middleware.

Reply:
1:  This is being handled in the front end and, if so the ghLoginModal is called and prompts the user to log in!
2: If the token is not valid the validateTeamRepo from the middleware catches this and exits early ,the code doesn't reach this function
3: This particular line is being duplicated to retrieve the username from the ghToken , the solution to this would be sending the username and reponame as part of the body , but for this i would have to make an api call from the client /front end.